### PR TITLE
feat: added createdAt and updatedAt columns in user entity

### PR DESCRIPTION
**Pull request by Saad-Amjad**

### CODE REVIEW

File: src/user/entities/user.entity.ts [hunk 1]
```diff
@@ -24,4 +32,10 @@ export class User {
   @Unique('email', ['email'])
   @Column({ length: 200 })
   email: string;
+
+  @CreateDateColumn({ name: 'created_at', nullable: true })
```
Comment (yashmurty) [inline src/user/entities/user.entity.ts:36]:
```suggestion
  @CreateDateColumn({ nullable: true })
```
@Saad-Amjad It's okay to use `camelCase` for column names in our tables. 
We are doing the same for all other existing columns as well. :bow:

**Review by yashmurty — CHANGES_REQUESTED** [review]

File: src/user/entities/user.entity.ts (hunk 1 above)
Comment (yashmurty) [reply src/user/entities/user.entity.ts:36]: Also @Saad-Amjad I just realized, we will need to create `migration` files for this change.
Let me ask @sivanpayyadakath to take over this PR. :bow:

File: test/test-utils.ts [hunk 2]
```diff
@@ -78,7 +78,7 @@ export const seedAdminUser = async (
 
   const authTokenForAdmin: AuthTokenOutput = loginResponse.body.data;
 
-  const adminUser: UserOutput = { ...userOutput };
+  const adminUser: UserOutput = JSON.parse(JSON.stringify(userOutput));
```
Comment (yashmurty) [inline test/test-utils.ts:81]: what's happening here?

File: src/user/entities/user.entity.ts [hunk 3]
```diff
@@ -24,4 +32,10 @@ export class User {
   @Unique('email', ['email'])
   @Column({ length: 200 })
   email: string;
+
+  @CreateDateColumn({ name: 'createdAt', nullable: true })
+  createdAt: Timestamp;
```
Comment (yashmurty) [inline src/user/entities/user.entity.ts:37]:
```suggestion
  createdAt: Timestamp;
```
We can keep this as `string` as well if need be. :thinking:

File: test/auth/auth.e2e-spec.ts [hunk 4]
```diff
@@ -51,21 +51,29 @@ describe('AuthController (e2e)', () => {
...
     };
+
     const registerOutput: RegisterOutput = {
       id: 2,
       name: 'e2etester',
       username: 'e2etester',
       roles: [ROLE.USER],
       isAccountDisabled: false,
       email: 'e2etester@random.com',
+      createdAt: null,
```
Comment (yashmurty) [inline test/auth/auth.e2e-spec.ts:62]: Does it return `null` when you do POST request?
Can you check if we can return the new values that are set on the DB by doing a `refresh: true` flag if necessary. :thinking:

Comment (sivanpayyadakath) [reply test/auth/auth.e2e-spec.ts:62]: @yashmurty 
> Does it return null when you do POST request?
No

I do this in following lines 
```
          registerOutput.createdAt = resp.data.createdAt
          registerOutput.updatedAt = resp.data.updatedAt
```

File: test/auth/auth.e2e-spec.ts [hunk 5]
```diff
@@ -51,21 +51,29 @@ describe('AuthController (e2e)', () => {
...
       return request(app.getHttpServer())
         .post('/auth/register')
         .send(registerInput)
         .expect(HttpStatus.CREATED)
-        .expect({ data: registerOutput, meta: {} });
+        .expect((res) => {
+          const resp = res.body;
+          registerOutput.createdAt = resp.data.createdAt;
+          registerOutput.updatedAt = resp.data.updatedAt;
+          expect(resp).toEqual({ data: registerOutput, meta: {} });
```
Comment (yashmurty) [inline test/auth/auth.e2e-spec.ts:75]: @sivanpayyadakath can you update this to use `objectContaining` test instead?

**Review by yashmurty — APPROVED** [review]
LGTM! :tada:

**Pull request merged**